File: examples/go/enum-color/color.go
// Package color is the str-pjlc1 end-to-end fixture for enum value-domain
// extraction. `Color` is a named string type ("enum") with a three-value const
// set; `ClassifyColor` switches over all three plus a default arm. Without a
// value domain the generator only produces generic strings ("a"/"hello") that
// all fall to the default arm, so the three valid arms are never reached. With
// enum_values carried on the param's union TypeInfo, the core draws RED/GREEN/
// BLUE and reaches every arm.
package color

// Color is a named string enum with a constant set.
type Color string

// The three-value const set that forms Color's value domain.
const (
	ColorRed   Color = "RED"
	ColorGreen Color = "GREEN"
	ColorBlue  Color = "BLUE"
)

// ClassifyColor switches over the Color enum with an explicit default arm.
// Four reachable return values: the three valid members plus the off-domain
// default probe.
func ClassifyColor(c Color) string {
	switch c {
	case ColorRed:
		return "warm"
	case ColorGreen:
		return "cool-green"
	case ColorBlue:
		return "cool-blue"
	default:
		return "invalid"
	}
}

File: examples/rust/enum-color/src/color.rs
//! str-2nfoe end-to-end fixture for Rust enum value-domain extraction.
//!
//! `Color` is a fieldless enum with a three-member value domain; `classify_color`
//! matches over all three variants. Without a value domain the core generator
//! only produces generic strings that all fail to deserialize into `Color`, so
//! no arm is ever reached. With `enum_values` carried on the param's union
//! `TypeInfo` (variant names "Red"/"Green"/"Blue"), the generator draws valid
//! members and reaches every arm.

use serde::{Deserialize, Serialize};

/// A fieldless enum whose default-serde value domain is its variant names.
#[derive(Serialize, Deserialize)]
pub enum Color {
    Red,
    Green,
    Blue,
}

/// Match over the `Color` enum, one distinct return per variant.
///
/// The match is exhaustive (no default arm): a fieldless enum with all variants
/// covered leaves no residual case, and an off-domain string simply fails to
/// deserialize into `Color` before the body runs.
pub fn classify_color(c: Color) -> &'static str {
    match c {
        Color::Red => "warm",
        Color::Green => "cool-green",
        Color::Blue => "cool-blue",
    }
}
